src/Gaufrette/Adapter/GoogleCloudClientStorage.php

```diff
@@ -0,0 +1,295 @@
+        } catch (\Exception $e) {
+            throw StorageFailure::unexpectedFailure('write', ['key' => $key, 'content' => $content], $e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists($key)
+    {
+        return $this->bucket->object($this->computePath($key))->exists();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isDirectory($key)
+    {
+        return $this->exists($this->computePath(rtrim($key, '/')).'/');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function listKeys($prefix = null)
+    {
+        // @FIXME : the list should not return the root directory in the keys
+        $keys = array();
+
+        foreach ($this->bucket->objects(array('prefix' => $this->computePath($prefix))) as $e) {
+            $keys[] = strlen($this->options['directory'])
+                ? substr($e->name(), strlen($this->options['directory'] . '/'))
+                : $e->name()
+            ;
+        }
+
+        sort($keys);
+
+        return $keys;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keys()
+    {
+        return $this->listKeys();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function mtime($key)
+    {
+        $info = $this->bucket->object($this->computePath($key))->info();
+
+        return strtotime($info['updated']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($key)
+    {
+        try {
+            $this->bucket->object($this->computePath($key))->delete();
+        } catch (\Exception $e) {
+            if ($e instanceof NotFoundException) {
+                throw new FileNotFound($key);
+            }
+
+            throw StorageFailure::unexpectedFailure('delete', ['key' => $key], $e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function rename($sourceKey, $targetKey)
+    {
+        $pathedSourceKey = $this->computePath($sourceKey);
+        $pathedTargetKey = $this->computePath($targetKey);
+
+        try {
+            $object = $this->bucket->object($pathedSourceKey);
+            $metadata = $this->getMetadata($sourceKey);
+
+            $copy = $object->copy($this->bucket,
+                array(
+                    'name' => $pathedTargetKey
+                )
+            );
+
+            $this->setAcl($copy);
+            $this->setMetadata($targetKey, $metadata);
+
+            $object->delete();
+        } catch (\Exception $e) {
+            if ($e instanceof NotFoundException) {
+                throw new FileNotFound($key);
+            }
+
+            throw StorageFailure::unexpectedFailure('rename', ['sourceKey' => $sourceKey, 'targetKey' => $targetKey], $e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMetadata($key)
+    {
+        try {
+            $infos = $this->bucket->object($this->computePath($key))->info();
+
+            return isset($infos['metadata'])
```

Comment:
Maybe use the coalesce operator here :thinking: (and maybe use a cs-fixer to fix everything :stuck_out_tongue:)